doc/sphinx/addendum/generalized-rewriting.rst

```diff
@@ -759,10 +759,49 @@ example of a mostly user-space extension of the algorithm.
 Constant unfolding
 ~~~~~~~~~~~~~~~~~~
 
-The resolution tactic is based on typeclasses and hence regards user-defined
-:term:`constants <constant>` as transparent by default. This may slow down the
-resolution due to a lot of unifications (all the declared ``Proper``
-instances are tried at each node of the search tree). To speed it up,
+.. cmd::  Hint {| Variables | Constants } {| Opaque | Transparent } : rewrite.
+
+   By default, :tacn:`setoid_rewrite` and :tacn:`rewrite_strat` unfold global
+   :term:`definitions <constant>` during rewrite-rule matching, but do not
+   unfold local definitions. The former may slow down matching, whereas the
+   latter may cause matches to be missed. This behavior is configurable using
+   transparency hints in the special hint database ``rewrite`` as shown,
+   similarly to the effect of :cmd:`Hint Transparent` on unification used by
+   :tacn:`eauto` and related tactics.
+
+   .. coqtop:: reset in
+
+      Require Setoid.
+      Definition f x y := 2*x + y.
+      Definition g x y := 2*x + y.
+
+      Goal forall
+        (double_f : forall x y, 2*f x y = f (2*x) y + y),
+        2 * g 1 8 = 20.
+      Proof.
+        intros.
+          assert_succeeds (setoid_rewrite double_f).
+          assert_succeeds (rewrite_strat bottomup double_f).
+        set (x :=g _).
+          assert_fails (setoid_rewrite double_f).
+          assert_fails (rewrite_strat bottomup double_f).
+        Hint Variables Transparent : rewrite.
+          assert_succeeds (setoid_rewrite double_f).
+          assert_succeeds (rewrite_strat bottomup double_f).
+        Hint Constants Opaque : rewrite.
+          assert_fails (setoid_rewrite double_f).
+          assert_fails (rewrite_strat bottomup double_f).
+        subst x.
+          assert_fails (setoid_rewrite double_f).
+          assert_fails (rewrite_strat bottomup double_f).
+        exact eq_refl.
+      Qed.
+
+The resolution of ``Proper`` instances is based on :tacn:`typeclasses eauto`
+and hence regards user-defined :term:`constants <constant>` as transparent by
+default. This may slow down the resolution due to a lot of unifications (all
+the declared ``Proper`` instances are tried at each node of the search tree).
+To speed it up,
 declare your constant as rigid for proof search using the command
```

Comment:
If "rigid" means "opaque", please use that.  Otherwise I don't know what it means.